pkg/sql/distsqlrun/server.go

```diff
@@ -95,28 +99,65 @@ type ServerConfig struct {
 // ServerImpl implements the server for the distributed SQL APIs.
 type ServerImpl struct {
 	ServerConfig
-	flowRegistry  *flowRegistry
-	flowScheduler *flowScheduler
-	memMonitor    mon.MemoryMonitor
-	regexpCache   *parser.RegexpCache
+	flowRegistry      *flowRegistry
+	flowScheduler     *flowScheduler
+	memMonitor        mon.MemoryMonitor
+	regexpCache       *parser.RegexpCache
+	engine            *engine.RocksDB
+	processorRegistry *uint64
 }
 
 var _ DistSQLServer = &ServerImpl{}
 
 // NewServer instantiates a DistSQLServer.
 func NewServer(ctx context.Context, cfg ServerConfig) *ServerImpl {
+	// TODO(arjun): set sane defaults
+	rocksDBCache := engine.NewRocksDBCache(0)
```

Comment:
I would extract rocksDB creation to a separate function so that we can use it in tests.